scanner/decisions/0010-separate-versioning-for-vuln-bundles.md

```diff
@@ -0,0 +1,82 @@
+|--------|--------------|--------------------------------------------------------------------|
+| dev    | heads/master | This stream tracks the latest vulnerability schema                 |
+| v1     | heads/4.4.5  | This stream tracks the schema used by the current Stackrox release |
+
+On the subsequent releases (for example,. `4.4.6`), one of the two will be carried out:
+
+1.  The schema hasn't changed, so v1 now tracks the new release (`heads/4.4.6`).
+2.  The schema has changed, so v1 stays at current release (`heads/4.4.5`), and a new version, v2, is created to track the next release (`heads/4.4.6`).
+
+### Building bundle version in Exporter
+
+The GitHub workflow will have a map with "bundle version" as keys, and "git reference" as values, similar to table 1 above.  This can be stored statically in the matrix definition.
+
+For each version in the mapping, a bundle will be built based on the specified git ref and stored at `https://definitions.stackrox.io/v4/<bundle-version>/vulnerabilities.zip`.
+
+In every release cut, the key specified in `scanner/VULNERABILITY_VERSION` will be updated in the mapping to the current release branch reference (e.g., `heads/4.4.6`).  Notice that if the version is the same, it will be updated.  If it's new, it will be added, and the previous version will stay.
+
+### Selection bundle version in Scanner
+
+Scanner will determine which version bundle it wants based on hard-coded information instead during build time.  A file at `scanner/VULNERABILITY_VERSION` will determine the version on release builds and nightlies.  On non-release builds this will be set to "dev".
+
+Today scanner supports a special variable `ROX_VERSION` in the vulnerability URL configuration.  A new variable will be added called `ROX_SCANNER_VULNERABILITY_VERSION` to specify the expected bundle version to fetch.
+
+When a schema bump is detected (e.g., a ClairCore bump that brings in incompatible schema changes to the vulnerability structs), this file should be updated.
+
+The definitions handler in Central can be kept the same.  It already supports arbitrary versions in the GET request from Scanner.
+
+[The check for nightlies](https://github.com/stackrox/stackrox/blob/e332cd19a639f59d9931414a4f4e561981396dad/central/scannerdefinitions/handler/handler.go#L661) is unnecessary and will be removed.
+
+### Offline mode
+
+The offline mode will continue to serve every supported bundle version.  But instead of offering the latest Y-stream, it will offer all bundles supported by the vulnerability workflow, except for "dev".
+
+### The bundle ingestions
+
+The bundle will modified to contain as `meta.json` file, which will carry the stackrox version and bundle version used to build it.  Scanner will include additional logic to reject a bundle that is not at the expected version.
+
+### Deprecation
+
+Once a Stackrox release is out of support, we will remove the entry from the mapping, effectively disabling updates for that stream.
```

Comment:
This will need a grace period or something like 'AFTER a release has been out of support for two cycles, then it will stop getting updates', it takes time for users validate a new version and upgrade.

Reply:
I agree. We should come up with a formal policy and then add it to the product documentation, so nobody is (allowed to be) surprised (if they read the docs).

Reply:
I added a comment to the ADR to open room for such policy, but let's define it elsewhere (or even leave it open so we have some flexibility to change it easily).